corehq/util/metrics/__init__.py

```diff
@@ -0,0 +1,15 @@
+from django.utils.functional import SimpleLazyObject
+
+from corehq.util.metrics.datadog import DatadogMetrics
+from corehq.util.metrics.metrics import DummyMetrics, DelegatedMetrics, HqMetrics
+from corehq.util.metrics.prometheus import PrometheusMetrics
+
+
+def _get_metrics():
+    enabled = list(filter(lambda m: m.enabled(), [DatadogMetrics(), PrometheusMetrics()]))
+    if not enabled:
+        return [DummyMetrics()]
+    return enabled
+
+
+metrics = DelegatedMetrics(SimpleLazyObject(_get_metrics))  # singleton/global
```

Comment:
Initially I had this as a function `get_metrics()` but after making it lazy there wasn't a need for the function. This will only call the `_get_metrics` function when a metric class is actually used. This is needed make sure that Django is fully set up before determining if the metrics providers are enabled (since they rely on settings).

```python
m = metrics.counter('name', 'description')  # still lazy
m.inc()  # now it's evaluated
```

Reply:
Does this mean that `DelegatedMetrics` is always used even when there is only a single "metrics" object to delegate to? That seems wasteful. Would it be possible to only delegate if there is more than one (I would expect that to be the exceptional case).

Reply:
Yes it does mean that. The alternative is to split `DelegatedMetrics` into two classes: `DelegatedMetrics` and `LazyMetrics` and then combine them as needed (the laziness is always needed).